tests/resources/test_file_link.py

```diff
@@ -92,25 +93,22 @@ def test_delete(collection, session):
     assert expected_call == session.last_call
 
     # A URL that does not follow the files/{id}/versions/{id} format is invalid
-    invalid_url = 'www.citrine.io/develop/filestuff/{}'.format(file_id)
-    invalid_file_link = collection.build(FileLinkDataFactory(url=invalid_url))
-    with pytest.raises(AssertionError):
-        collection.delete(invalid_file_link)
-
+    for chunk in (f'{file_id}', f'{file_id}/{version_id}'):
+        invalid_url = f'{collection._get_path}/{chunk}'
+        invalid_file_link = collection.build(FileLinkDataFactory(url=invalid_url))
+        with pytest.raises(ValueError):
+            collection.delete(invalid_file_link)
 
-@patch('citrine.resources.file_link.boto3_client')
-@patch('citrine.resources.file_link.open')
-@patch('citrine.resources.file_link.os.stat')
-@patch('citrine.resources.file_link.os.path.isfile')
-def test_upload(mock_isfile, mock_stat, mock_open, mock_boto3_client, collection, session):
-    """Test signaling that an upload has completed and the creation of a FileLink object."""
-    StatStub = namedtuple('StatStub', ['st_size'])
+    # A remote URL is invalid
+    ext_invalid_url = f'http://www.citrine.io/develop/files/{file_id}/versions/{version_id}'
+    ext_invalid_file_link = collection.build(FileLinkDataFactory(url=ext_invalid_url))
+    with pytest.raises(ValueError):
+        collection.delete(ext_invalid_file_link)
 
-    mock_isfile.return_value = True
-    mock_stat.return_value = StatStub(st_size=22300)
-    mock_open.return_value.__enter__.return_value = 'Random file contents'
-    mock_boto3_client.return_value = FakeS3Client({'VersionId': '3'})
```

Comment:
All of this was actually testing implementation details that got broken in the swap over to pathlib.